test/index.test.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+/* eslint-disable no-console*/
+const assert = require('chai').assert;
+const sinon = require('sinon');
+const mockery = require('mockery');
+
+sinon.assert.expose(assert, { prefix: '' });
+
+const TEST_XML =
+`<project>
+<description/>
+<keepDependencies>false</keepDependencies>
+<properties/>
+<scm class="hudson.scm.NullSCM"/>
+<canRoam>true</canRoam>
+<disabled>false</disabled>
+<blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+<blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+<triggers/>
+<concurrentBuild>false</concurrentBuild>
+<builders>
+<hudson.tasks.Shell>
+<command>sleep 100 | echo 'Hello, I am going to sleep for 100s!'</command>
+</hudson.tasks.Shell>
+</builders>
+<publishers/>
+<buildWrappers/>
+</project>`;
+
+describe('index', () => {
+    let executor;
+    let Executor;
+    let fsMock;
+    let jenkinsMock;
+    let breakerMock;
+    let createOpts;
+    let buildOpts;
+
+    /**
+     * Stub for Breaker factory method
+     */
+    function BreakerFactory() {}
+
+    const config = {
+        buildId: '4b8d9b530d2e5e297b4f470d5b0a6e1310d29c5e',
+        jobId: '50dc14f719cdc2c9cb1fb0e49dd2acc4cf6189a0',
+        pipelineId: 'ccc49349d3cffbd12ea9e3d41521480b4aa5de5f',
+        container: 'node:4',
+        scmUrl: 'git@github.com:screwdriver-cd/data-model.git#master'
+    };
+
+    const jobName = config.buildId;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        fsMock = {
+            readFile: sinon.stub()
+        };
+
+        jenkinsMock = {
+            job: {
+                create: sinon.stub(),
+                build: sinon.stub()
+            }
+        };
+
+        breakerMock = {
+            runCommand: sinon.stub()
+        };
+
+        mockery.registerMock('fs', fsMock);
+    });
+
+    afterEach(() => {
+        mockery.deregisterAll();
+        mockery.resetCache();
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('extends base class', () => {
+        // eslint-disable-next-line global-require
+        Executor = require('../index');
+
+        executor = new Executor({
+            host: 'jenkins',
+            username: 'admin',
+            password: 'fakepassword'
+        });
+
+        // eslint-disable-next-line global-require
+        const BaseExecutor = require('../node_modules/screwdriver-executor-base/index');
+
+        assert.instanceOf(executor, BaseExecutor);
+    });
+
+    describe('start', () => {
+        beforeEach(() => {
+            BreakerFactory.prototype.runCommand = breakerMock.runCommand;
```

Comment:
I usually find that using `className.prototype.*` in setting up mocks indicates an issue. You can usually setup the "factory" as `sinon.stub()` and have it return an implementation. This also allows you to test the "new" calls to the factory itself.

In this case:
```
BreakerFactory = sinon.stub().returns(breakerMock);
```